pkg/ipamd/ipamd.go

```diff
@@ -575,35 +574,28 @@ func (c *IPAMContext) nodeInit() error {
 
 	eniConfigName, err := eniconfig.GetNodeSpecificENIConfigName(node)
 	if err == nil && c.useCustomNetworking && eniConfigName != "default" {
-		// Signal to VPC Resource Controller that the node is using custom networking
-		err := c.SetNodeLabel(ctx, vpcENIConfigLabel, eniConfigName)
+		// Add the feature name to CNINode of this node
+		err := sgpp.AddFeatureToCNINode(ctx, c.cachedK8SClient, c.myNodeName, rc.CustomNetworking, eniConfigName)
```

Comment:
Do we really want to crash the CNI? Can't we simply retry and emit a log each time? Crashing will subject us to crash loop backoff durations.

Reply:
If Get returns error of NotFound, we just log the error and move on. The reconciler will try again.

Reply:
In this implementation, for custom networking, it looks like we only patch the CRD during node init. So if we cannot patch during node init, it would never be patched, and if the k8s client returns an error, `aws-node` pod will crash.

It looks like only SGPP uses a reconciler approach

Reply:
I don't think we support reconciling on the eniconfig label currently. Can you double check? I checked the code in master branch and seems we only do that in init.

Reply:
Right, we do not currently support reconciling on ENIConfig label, so this is not a change in behavior, I am just pointing out that our restart strategy here is to crash the `aws-node` pod when the CRD does not exist. Which may not be the best idea as @ellistarn pointed out, but it's the least complex

Reply:
Isn't it straightforward for us to write a controller-runtime reconciler that's included with the ipamd server?

Reply:
If you mean to reconcile on the CRD, then yeah, but I think that overhead is not worth it. We have a push to decrease the amount of API server calls from IPAMD: https://github.com/aws/amazon-vpc-cni-k8s/issues/2160

Reply:
ack, this is a good idea for daemons. As an alternative we could backoff-poll to retrieve the CRD instead of crashing the process.

Reply:
For reference, when we refer to the "reconciliation loop", it refers to this: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/ipamd/ipamd.go#L685

This is the "main loop" for IPAMD, where we continuously check if we can reclaim IPs, need to attach new ENIs, etc.